lib/utils/expandObjectTokens.js

```diff
@@ -178,11 +178,22 @@ export function expandToken(token, opts, platform) {
       expandedTokenObjRef = expandedTokenObjRef[index + 1];
     }
     Object.entries(objectVal).forEach(([key, value]) => {
-      expandedTokenObjRef[key] = {
-        ...copyMeta,
+      expandedTokenObjRef[key] = deepmerge(copyMeta, {
         [`${uses$ ? '$' : ''}value`]: value,
         [`${uses$ ? '$' : ''}type`]: getTypeFromMap(key, compositionType, typesMap),
-      };
+        // store the original composite token type as well as which prop it was
+        // so that for example a lineHeight property of a typography token, when expanded
+        // can still be matched as a lineHeight token, even though its type is now "number" after the expansion
+        $extensions: {
```

Comment:
This is the first time Style Dictionary is using `$extensions`. Do we expect to put more in here later?

Reply:
I'm thinking that perhaps it makes sense to move other properties that Style Dictionary adds to the token metadata under this namespace, as not to clash with metadata that the token author themselves may add. This could be planned for a future v5 I reckon